### DISCUSSION TITLE
useFileSystemAccess - how to save file on the server?

### DISCUSSION THREAD

**Discussion by mihail727** [vueuse/vueuse#1474] [Q&A] [answered]
i'm trying to upload file from client to server and then save it locally on the server.
i'm using VueUse as Nuxt 3.
`@/pages/upload.vue`
```vue
<script setup lang="ts">
	const onChange = async (e: Event) => {
		const { data, file, open } = useFileSystemAccess({ dataType: 'Text' });

		await open();

		await useRequest('/api/upload/image', {
			method: 'POST',
			body: {
				fileName: file.value.name,
				fileData: data.value,
			},
		});
	};
</script>

<template>
	<div>
		<button @click="onChange">Click</button>
	</div>
</template>
```
`@/server/api/upload.ts`
```ts
import { createApp, createRouter, useQuery, useBody } from 'h3';

const app = createApp();
const router = createRouter();

app.use(router);

router.post('/image', async (req, res): Promise<IServerResponse<any>> => {
	try {
		const { fileName, fileData } = await useBody(req);
		fs.writeFileSync(fileName, fileData);
		return { success: true };
	} catch (e) {
		return { success: false, reason: e.message };
	}
});

export default app;
```
---
### but i need upload .pdf, .png files.... and dataType can have values: `Text`, `Blob`, `ArrayBuffer` how to parse it on the server to be able save it?

**Comment by wheatjs** [comment]
`useFileSystemAccess` only gives access to the users local file system, not the server. To upload a file to the server you will need to do something like make an endpoint on the server to accept a file upload and then use fetch or something and upload the file

**Reply by mihail727** [reply]
Yes, I understand, but I do not understand how to process it on the server
I tried with `formdata`, but when I try to parse it on the server, I get the error `'bad content-type header, no multipart boundary'`
although here I suspect that useFetch from nuxt 3 is to blame
I also tried to send directly to the server, without `formdata`, using `Blob` or `ArrayBuffer`, but here I simply do not understand how to convert these formats to a file - I managed to create a file, but it was empty

**Comment by mihail727** [comment, marked as answer]
my solution [here](https://github.com/nuxt/framework/discussions/4051)
